File: src/main/java/mods/eln/gridnode/electricalpole/ElectricalPoleDescriptor.java
package mods.eln.gridnode.electricalpole;

import mods.eln.gridnode.GridDescriptor;
import mods.eln.misc.Obj3D;
import mods.eln.misc.VoltageLevelColor;
import mods.eln.sixnode.electricalcable.ElectricalCableDescriptor;

/**
 * Created by svein on 07/08/15.
 */
public class ElectricalPoleDescriptor extends GridDescriptor {
    public final float minimalLoadToHum = 0.2f;
    public final boolean includeTransformer;

    public ElectricalPoleDescriptor(String name, Obj3D obj, String cableTexture, ElectricalCableDescriptor cableDescriptor, boolean includeTransformer) {
        super(name, obj, ElectricalPoleElement.class, ElectricalPoleRender.class, cableTexture, cableDescriptor);
        this.includeTransformer = includeTransformer;
        if (includeTransformer) {
            rotating_parts.add(obj.getPart("transformer"));
            rotating_parts.add(obj.getPart("cables"));
            static_parts.add(obj.getPart("foot"));
        }

        changeDefaultIcon(name.replaceAll("/", ""));
        voltageLevelColor = VoltageLevelColor.Grid;
    }
}

File: src/main/java/mods/eln/gridnode/electricalpole/ElectricalPoleDescriptor.kt
package mods.eln.gridnode.electricalpole

import mods.eln.gridnode.GridDescriptor
import mods.eln.misc.Obj3D
import mods.eln.sixnode.electricalcable.ElectricalCableDescriptor

/**
 * Created by svein on 07/08/15.
 */
class ElectricalPoleDescriptor(name: String, obj: Obj3D, cableTexture: String, cableDescriptor: ElectricalCableDescriptor, val includeTransformer: Boolean) : GridDescriptor(name, obj, ElectricalPoleElement::class.java, ElectricalPoleRender::class.java, cableTexture, cableDescriptor) {
    val minimalLoadToHum = 0.2f

    init {
        obj.getPart("foot")?.let {
            static_parts.add(it)
        }
        if (includeTransformer) {
            arrayOf("trafo", "cables").forEach {
                obj.getPart(it)?.let { rotating_parts.add(it) }
            }
        }
    }
}
